CRM/Core/BAO/Address.php

```diff
@@ -1042,11 +1042,21 @@ public static function processSharedAddress($addressId, $params) {
     $query = 'SELECT id, contact_id FROM civicrm_address WHERE master_id = %1';
     $dao = CRM_Core_DAO::executeQuery($query, array(1 => array($addressId, 'Integer')));
 
+    // TODO: VALIDATION REQUIRED = was implicit and is now explicit
+    // should we keep it backward compatible (i.e. no params means creating relationship) ?
+    // this should only impact API calls
```

Comment:
Answer: yes, we need to keep it backwards compatible, so if the param is not set, treat it as if it were passed in TRUE.